src/main/java/teammates/ui/controller/InstructorStudentRecordsPageData.java

```diff
@@ -1,24 +1,99 @@
+    private String courseId;
+    private String studentName;
+    private String studentEmail;
+    private String googleId;
+    private String showCommentBox;
+    private StudentProfile studentProfile;
+    public StudentProfileAttributes spa; // used for testing admin message
+    private List<CommentsForStudentsTable> commentsForStudentTable;
+    private List<String> sessionNames;
 
-    public InstructorStudentRecordsPageData(AccountAttributes account) {
-        super(account);
+    public InstructorStudentRecordsPageData(AccountAttributes account, StudentAttributes student,
+                                            String courseId, String showCommentBox, StudentProfileAttributes spa,
+                                            List<CommentAttributes> comments, List<String> sessionNames,
+                                            InstructorAttributes instructor) {
+        super(account, student);
+        this.courseId = courseId;
+        this.studentName = Sanitizer.sanitizeForHtml(student.name);
+        this.studentEmail = student.email;
+        this.googleId = account.googleId;
+        this.showCommentBox = showCommentBox;
+        if (spa != null) {
+            this.spa = spa;
+            String pictureUrl = Const.ActionURIs.STUDENT_PROFILE_PICTURE
+                              + "?" + Const.ParamsNames.BLOB_KEY + "=" + spa.pictureKey
+                              + "&" + Const.ParamsNames.USER_ID + "=" + account.googleId;
+            if (spa.pictureKey.isEmpty()) {
+                pictureUrl = Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH;
+            }
+            this.studentProfile = new StudentProfile(this.studentName, spa, pictureUrl);
+        }
+        List<Comment> commentDivs = new ArrayList<Comment>();
+        for (CommentAttributes comment : comments) {
+            Comment commentDiv = new Comment(comment, "You", student.name + " (" + student.team + ", " + student.email + ")");
+            String whoCanSeeComment = getTypeOfPeopleCanViewComment(comment);
+            commentDiv.setVisibilityIcon(whoCanSeeComment);
+            commentDiv.setEditDeleteEnabled(false);
+            commentDiv.setNotFromCommentsPage(student.email);
+            commentDiv.setNumComments(comments.size());
+            commentDivs.add(commentDiv);
+        }
+        this.commentsForStudentTable = new ArrayList<CommentsForStudentsTable>();
+        CommentsForStudentsTable commentsForStudent = new CommentsForStudentsTable("You", commentDivs);
+        commentsForStudent.setInstructorAllowedToGiveComment(instructor.isAllowedForPrivilege(student.section,
+                                        Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS));
+        commentsForStudentTable.add(commentsForStudent);
+        this.sessionNames = sessionNames;
+    }
+
+    public String getStudentEmail() {
+        return studentEmail;
+    }
+
+    public String getGoogleId() {
+        return googleId;
+    }
```

Comment:
Is storing it in the class required?
`account.googleId` should work fine as well. (or `${data.account.googleId}`)